lib/mapping/CMap.cpp

```diff
@@ -343,9 +347,15 @@ bool CMap::checkForVisitableDir(const int3 & src, const TerrainTile *pom, const
 
 		const CGObjectInstance * obj = pom->visitableObjects[b];
 
-		if (!obj->appearance.isVisitableFrom(src.x - dst.x, src.y - dst.y))
+		if(!obj->appearance.isVisitableFrom(src.x - dst.x, src.y - dst.y))
 			return false;
 	}
+
+	// TODO: This hacky code need improvement. It's used to not let hero exit from teleporter to wrong tile. E.g above teleporter.

```

Comment:
It's needed because currently if hero staying on teleporter CGameState::getNeighbours always return all non-blocked tiles around teleporter as neighbours. Originally I there was workaround (okay, just dirty hack) for that problem in CPathfinder::canMoveBetween that check pathability in both directions.

Of course to make sure unidirectional teleporters work I had to remove hack out of canMoveBetween and find out about that problem with teleporters.

So I need to find out what wrong with getNeighbours and then I'll be able to get rid of it.